Automattic/woocommerce-services
# Add notice about disconnection when uninstalling Jetpack

## Description

Now that Jetpack will no longer be required by WCS&T, it could be uninstalled from a site. However when that happens, Jetpack will automatically remove its tokens from a site, therefore disconnecting WCS&T from WPCOM.

Although we will display our good old "Connect your site" banners after a page refresh/navigating to a page where these banners are displayed (e.g. Plugins page, WC Settings page, Edit Order page) so merchants using label printing might not miss this, this might be an issue for users of the tax rates or live shipping rates.

To make sure merchants keep their sites connected, we're adding a notice after Jetpack is uninstalled.

The notice will only show up after Jetpack is uninstalled, and only if WCS&T has been connected. It will also not allow connecting if already connected thanks to leveraging the NUX flow's `register_woocommerce_services_jetpack` action that is only registered when the `before_jetpack_connection` banner should be displayed.

### Related issue(s)

Fixes https://github.com/woocommerce/shipping/issues/84

### Steps to reproduce & screenshots/GIFs

https://github.com/Automattic/woocommerce-services/assets/1759681/80e15010-c980-48c3-8807-ebaf2ea7c783

### Testing instructions

1. Set up your local environment as described in this PR: https://github.com/Automattic/woocommerce-services/pull/2439 (note the Woo-branded WPCOM auth page is only available when sandboxed - that PR has information on this)
2. Make sure your site is connected to WPCOM.
3. Reproduce the steps in the video above.
